src/_guides/language/concurrency/index.md

```diff
@@ -0,0 +1,403 @@
+This Dart code might register some event listeners —
+to respond to user input or file I/O, for example.
+When the isolate's initial function returns,
+the isolate stays around if it needs to handle events.
+After handling the events, the isolate exits.
+
+![A more general figure showing that any isolate runs some code, optionally responds to events, and then exits](/guides/language/concurrency/images/basics-isolate.png)
+
+In a client app, the main isolate’s event queue might contain
+repaint requests and notifications of tap and other UI events, like this:
+***[PENDING: describe the figure in more detail.]***
+
+![PENDING: alt text here](/guides/language/concurrency/images/event-loop.png)
+
+The event handling happens on the main isolate after `main()` exits:
+***[PENDING: describe the figure in more detail.]***
+
+![PENDING: alt text here](/guides/language/concurrency/images/event-handling.png)
+
+If a synchronous operation takes too much processing time,
+the app can become unresponsive.
+In the following figure, the tap-handling code takes too long,
+so subsequent events are handled too late:
+***[PENDING: describe the figure in more detail.]***
+
+![PENDING: alt text here](/guides/language/concurrency/images/event-jank.png)
+
+In client apps, the result of a too-lengthy synchronous operation is often
+[janky (non-smooth) UI animation][jank].
+Worse, the UI might become completely unresponsive.
+
+[jank]: https://docs.flutter.dev/perf/rendering
+
+
+### Using a single-task background worker
+
+If your app’s UI becomes unresponsive due to a time-consuming computation —
+[parsing a large JSON file][json], for example —
+consider offloading that computation to a background worker.
+A common case, which is supported by the [Flutter `compute()` function][],
+is spawning an isolate that performs a computation and then exits:
+***[PENDING: describe the figure in a little more detail.]***
+
+[json]: https://docs.flutter.dev/cookbook/networking/background-parsing)
+ [Flutter `compute()` function]: https://docs.flutter.dev/cookbook/networking/background-parsing#4-move-this-work-to-a-separate-isolate
+
+![PENDING: alt text here](/guides/language/concurrency/images/isolate-bg-worker.png)
+
+A background worker can perform I/O
+(reading and writing files, for example), set timers, and more.
+It has its own memory (including its own copy of code) and
+doesn’t share any state with the main isolate.
+
+
+### Using the Isolate API
+
+Sometimes you might need to use the `Isolate` API directly.
+For example, you might use the `Isolate` API if
+your Dart Native app needs any of the following:
+
+* A background worker like what `compute()` supports,
+  but that might not be running in a Flutter app
+* A background worker that can
+  handle multiple requests or send multiple replies,
+  as shown in the following figure
+* Some other custom isolate behavior
+
+![PENDING: alt text here](/guides/language/concurrency/images/isolate-custom-bg-worker.png)
```

Comment:
I've been concentrating on _message passing_ rather than ports. For me, the whole port thing is a little harder to explain than the idea of message passing.

Reply:
I had a little something about messages just after the figure; I'm moving it up before the figure. We might want to add "message" to the Request & Reply arrows. Or add "request" and "reply" to the text in the custom background worker.

Reply:
OK, I think the current text works